modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/deployable/AbstractDeployableArtifactDescriptorFactory.java

```diff
@@ -129,6 +128,14 @@ protected void doDescriptorConfig(T descriptor) {
     if (!isMuleFramework()) {
       descriptor.setLogConfigFile(getLogConfigFile(getArtifactModel()));
     }
+
+    if (getArtifactModel().getSupportedJavaVersions() != null
+        && !getArtifactModel().getSupportedJavaVersions().isEmpty()) {
+      descriptor.setSupportedJavaVersions(getArtifactModel().getSupportedJavaVersions());
+    } else {
+      // If the model doesn't have this data assume the current jvm is good for this deployment.
+      descriptor.setSupportedJavaVersions(singleton(JAVA_SPECIFICATION_VERSION));
```

Comment:
We have this logic in a couple of places for setting the descriptor with a default value when the model doesn't have any version specified.
Wouldn't it make more sense to leave the descriptor without any version specified just like the model and then have the default logic in the validator?